lib/components/DynamicFonts/FontDownloader.ts

```diff
@@ -0,0 +1,144 @@
+import {Platform} from 'react-native';
+import RNFS from './RNFSPackage';
+import {FontExtension} from './FontLoader';
+
+const DEFAULT_DYNAMIC_FONTS_FOLDER = '/dynamicFonts';
+const DEFAULT_DOWNLOAD_ERROR_MESSAGE = 'An error occurred downloading the file:';
+
+export type FontDownloaderProps = {
+  /**
+   * Should be In the following convention: '/NAME'
+   */
+  dynamicFontsFolder?: string;
+  downloadErrorMessage?: string;
+  /**
+   * Enable debug mode to print extra logs
+   */
+  debug?: boolean;
+};
+
+// TODO: this can probably be a more general "downloader" if we so choose
+export default class FontDownloader {
+  private readonly props: FontDownloaderProps;
+  private readonly fs: NonNullable<typeof RNFS>;
+
+  constructor(props: FontDownloaderProps) {
+    this.props = {
+      dynamicFontsFolder: DEFAULT_DYNAMIC_FONTS_FOLDER,
+      downloadErrorMessage: DEFAULT_DOWNLOAD_ERROR_MESSAGE,
+      ...props
+    };
+
+    if (!RNFS) {
+      throw new Error(`RNUILib FontDownloader requires installing "react-native-fs" dependency`);
+    }
+    this.fs = RNFS;
+  }
+
+  private getPrivateFolder() {
+    const {dynamicFontsFolder} = this.props;
+    return (
+      (Platform.OS === 'android' ? this.fs.ExternalDirectoryPath : this.fs.DocumentDirectoryPath) + dynamicFontsFolder
```

Comment:
You don't need to add the 'dynamicFontsFolder' to the Android storage location? Does it not store on folders as in iOS?

Reply:
It adds it to both, there are parentheses